### PR TITLE
[test-only][PodSecurity] Scale testing of namespace bulk check behavior

### LINKED ISSUE #103303: [PodSecurity] Scale testing of namespace bulk check behavior
- [ ] Add tests to verify timeouts & pod limits on namespace checks

### PR DESCRIPTION
**Pull request by njuptlzf**
#### What type of PR is this?
/kind feature
/kind cleanup

#### What this PR does / why we need it:
Increase the coverage of `timeout` and `pods number` boundary
#### Which issue(s) this PR fixes:

Fixes #103303 

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
https://github.com/kubernetes/enhancements/tree/master/keps/sig-auth/2579-psp-replacement
```
/sig auth security

### CODE REVIEW

**Comment by njuptlzf** [comment]
need rebase after #103452

File: staging/src/k8s.io/pod-security-admission/admission/admission_test.go [hunk 1]
```diff
@@ -365,8 +377,28 @@ func TestValidateNamespace(t *testing.T) {
...
-		// TODO: test for bounding pod count with a warning
+		{
+			name:           "bounding evaluation time exceeded with a warning during update to restricted",
+			newLabels:      map[string]string{api.EnforceLevelLabel: string(api.LevelRestricted)},
+			oldLabels:      map[string]string{},
+			expectAllowed:  true,
+			expectListPods: true,
+			listDuration:   2 * time.Second,
+			expectEvaluate: api.LevelVersion{Level: api.LevelRestricted, Version: api.LatestVersion()},
+			expectWarnings: []string{"noruntimeclasspod: message", "Timeout reached after checking 1 pods"},
```
Comment (liggitt) [inline staging/src/k8s.io/pod-security-admission/admission/admission_test.go:388]: this is a surprising warning message... if our list timed out, I wouldn't have expected us to be able to check any pods

**Comment by liggitt** [comment]
this unit test looks complementary to https://github.com/kubernetes/kubernetes/pull/105457 which is an actual benchmark of how long it takes to verify an update to a namespace containing 3000 pods

**Comment by njuptlzf** [comment]
Sorry, I just saw it now. Unfortunately, I am participating in closed development, and people who are also interested are welcome to join.

**Pull request closed without merging**